mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -468,8 +468,10 @@ bool mav_vehicle::is_ready()
         return false;
     }
 
-    // Also check if mission waypoint has been received
+    // Also check if mission waypoint has been received. If it has not, we need
+    // to request it after takeoff.
     if (!get_mission_waypoint().is_initialized()) {
+        request_mission_item(this->mission_waypoint_id);
```

Comment:
I don't think this belong in a is_ready() function. This function should have no functionality other than checking for a set of conditions, this is beyond that scope.